extensions-api/src/main/scala/pl/touk/nussknacker/security/AuthCredentials.scala

```diff
@@ -1,3 +1,85 @@
 package pl.touk.nussknacker.security
 
-final case class AuthCredentials(value: String) extends AnyVal
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.Base64
+import javax.crypto.{Cipher, KeyGenerator, SecretKey}
+import scala.util.{Failure, Success, Try}
+
+sealed trait AuthCredentials
+
+object AuthCredentials {
+
+  val anonymousUsername     = "anonymous"
+  private val anonymousPass = "anonymous"
+
+  final case class PassedAuthCredentials(value: String) extends AuthCredentials
+  case object AnonymousAccess                           extends AuthCredentials
+
+  def fromStringToPassedAuthCredentials(value: String): AuthCredentials = {
+    PassedAuthCredentials(value)
+  }
+
+  def fromStringToAuthCredentials(value: String, crypter: Crypter): AuthCredentials = {
+    new String(Base64.getDecoder.decode(value)).split(":").toList match {
+      case `anonymousUsername` :: pass :: Nil =>
+        crypter.decrypt(pass) match {
+          case Failure(_) =>
+            PassedAuthCredentials(value)
+          case Success(decryptedAnonymousPassword) =>
+            decryptedAnonymousPassword match {
+              case `anonymousPass` => AnonymousAccess
+              case _               => PassedAuthCredentials(value)
+            }
+        }
+      case _ =>
+        PassedAuthCredentials(value)
+    }
+  }
+
+  implicit class Stringify(authCredentials: AuthCredentials) {
+
+    def stringify(crypter: Crypter): String = authCredentials match {
+      case PassedAuthCredentials(value) => value
+      case AnonymousAccess =>
+        new String(Base64.getEncoder.encode(s"$anonymousUsername:${crypter.encrypt(anonymousPass)}".getBytes))
+    }
+
+  }
+
+}
+
+trait Crypter {
```

Comment:
maybe should be `sealed`? Or do we want to leave extensibility?

Reply:
yeah, it can be sealed.